### PR TITLE
[rllib] Don't use flat weights in non-eager mode

### LINKED ISSUE #5983: Restoring a non-eager agent in eager mode throws an error
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
- **Ray installed from (source or binary)**: binary
- **Ray version**: 0.8.0dev6
- **Python version**: 3.6
- **Exact command to reproduce**:

Calling `agent._restore` on a saved checkpoint (which was trained without eager) with an `eager==True` config raises the error:

```python
~/anaconda3/envs/tensorflow_p36/lib/python3.6/site-packages/ray/rllib/policy/eager_tf_policy.py in set_weights(self, weights)
    357             variables = self.model.variables()
    358             assert len(weights) == len(variables), (len(weights),
--> 359                                                     len(variables))
    360             for v, w in zip(variables, weights):
    361                 v.assign(w)

AssertionError: (18691, 16)
```

### PR DESCRIPTION
**Pull request by ericl**
## Why are these changes needed?

This should allow the weights for an eager / non-eager model to be compatible.

## Related issue number

Closes https://github.com/ray-project/ray/issues/5983#issue-511330754

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

File: rllib/policy/tf_policy.py [hunk 1]
```diff
@@ -253,11 +251,16 @@ def learn_on_batch(self, postprocessed_batch):
 
     @override(Policy)
     def get_weights(self):
-        return self._variables.get_flat()
+        variables = self.model.variables()
+        return [v.numpy() for v in variables]
```
Comment (dmadeka) [inline rllib/policy/tf_policy.py]: Wouldnt this line only work if eager is enabled? Should it check for that?

Comment (dmadeka) [reply rllib/policy/tf_policy.py]: Confirming, training a model without eager gives a `numpy is only enabled in eager mode` error

Comment (ericl) [reply rllib/policy/tf_policy.py]: My bad, I think this should be fixed now. Try it again?

**Comment by ericl** [comment]
jenkins retest this please

**Comment by ericl** [comment]
Hm maybe this is breaking DQN somehow

**Comment by dmadeka** [comment]
I think so. ill try to look into it today

**Comment by ericl** [comment]
I think all tests are passing now. Ping @richardliaw for a careful review.

**Review by richardliaw — APPROVED** [review]
Is it possible that the custom policy has trainable variables outside of the model? Does the documentation need to reflect these changes?

**Pull request merged**